cmd/bnsd/x/username/username_test.go

```diff
@@ -77,6 +77,12 @@ func TestUsername(t *testing.T) {
 			WantName:   "😈",
 			WantDomain: "😀",
 		},
+		"invalid domain name": {
+			Raw:        "extreme*expert",
+			WantErr:    errors.ErrInput,
+			WantName:   "extreme",
+			WantDomain: "expert",
```

Comment:
q: why expect and verify this when an an error is returned? Just by reading the test it sets the wrong expectations that invalid data is still ok to use..

Reply:
This is because `ParseUsername` returns a username value regardless returned error value. Additionally, because you can manually cast string to a `Username`. Those methods can be used on a user name that is invalid.
Working with an invalid username is easy for the parser because the parsing rule is very simple. In addition, some invalid user names can be parsed without an error (ie `alice*example`).